File: Backend/App.DTO/Highlighted.cs
using System.ComponentModel.DataAnnotations;
using App.Domain;
using Base.Domain;

namespace App.DTO;

public class Highlighted : DomainEntityId
{
    public string Content { get; set; } = default!;

    public string? Comment { get; set; }
    
    public int OrderNo { get; set; }

    public int StartAt { get; set; }
    public int EndAt { get; set; }

    public Book? Book { get; set; }
    public Guid BookId { get; set; }

    public HighlightedType? HighlightedType { get; set; }
    public Guid HighlightedTypeId { get; set; }

    public ICollection<Like>? Likes { get; set; }

}

File: FrontEnd/src/Domain/Highlighted.ts
import type { IBook } from "@/Domain/IBook";
import type { IHighlightedType } from "@/Domain/HighlightedType";
import type { ILike } from "@/Domain/ILike";

export interface IHighlighted{
  id?: string,
  comment?: string,
  content: string,
  orderNo: number,
  startAt: number,
  endAt: number,

  book?: IBook,
  bookId?: string,
  highlightedType: IHighlightedType,
  likes: ILike[]
}